central/pod/datastore/datastore_impl.go

```diff
@@ -172,7 +175,16 @@ func (ds *datastoreImpl) RemovePod(ctx context.Context, id string) error {
 		sac.AllowFixedScopes(
 			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
 			sac.ResourceScopeKeys(resources.DeploymentExtension)))
-	return ds.indicators.RemoveProcessIndicatorsByPod(deleteIndicatorsCtx, id)
+
+	errIndicators := ds.indicators.RemoveProcessIndicatorsByPod(deleteIndicatorsCtx, id)
+
+	errPlop := ds.plops.RemovePlopsByPod(deleteIndicatorsCtx, id)
```

Comment:
Does it makes sense to go further if removing process indicators has failed? Would the state be consistent if PI removal will fail, but PLOP removal will succeed?

Reply:
> Does it makes sense to go further if removing process indicators has failed?

I think it does. I explain why below.

>  Would the state be consistent if PI removal will fail, but PLOP removal will succeed?

No, the listening endpoints would not be in an inconsistent state. The process indicators would be in an inconsistent state if removing them fails, but removing the listening endpoints won't make that situation worse.